### ISSUE TITLE
Pyright gives an error when passing a generic type argument with `Concatenate` to another generic class

### ISSUE THREAD

**Issue by StockerMC** [microsoft/pyright#2575]
**Describe the bug**
When passing a generic type argument to `Concatenate` to another generic class, the passed generic type(s) have their type lost.
This might be related to #2462, though I'm not sure.

**Expected behavior**
The following code to successfully type check.

**Screenshots or Code**
Note: this is in a stub file.
```python
from typing_extensions import Self, Concatenate, ParamSpec
from typing import Any, Callable, TypeVar, Protocol, Generic, overload

T = TypeVar('T', covariant=True)
O = TypeVar('O')
P = ParamSpec('P')

class _callable_alru_cache(Protocol[P, T]):
    foo: int
    def __call__(self, *args: P.args, **kwargs: P.kwargs) -> T:
        ...

class _wrapped_alru_cache(_callable_alru_cache[P, T], Generic[O, P, T]):
    # replacing _callable_alru_cache[Concatenate[O, P], T] with Callable[Concatenate[O, P], T] makes the error go away
    @overload
    def __get__(self, instance: None, owner: type[O]) -> _callable_alru_cache[Concatenate[O, P], T]: ...
    @overload
    def __get__(self, instance: O, owner: type[O]) -> Self: ...

@overload
def alru_cache(fn: Callable[Concatenate[A, P], T]) -> _wrapped_alru_cache[A, P, T]: ...
@overload
def alru_cache(fn: Callable[P, T]) -> _wrapped_alru_cache[Any, P, T]: ...


@alru_cache
def not_in_class(a: int, b: str) -> str: ...

class A:
    @alru_cache
    def in_class(self, a: int, b: str) -> str: ...

reveal_type(not_in_class) # _wrapped_alru_cache[Any, (a: int, b: str), str] - correct
not_in_class(1, '') # no error

a = A()

reveal_type(a.in_class) # _wrapped_alru_cache[A, (a: int, b: str), str] - correct
a.in_class(1, '') # no error

reveal_type(A.in_class) # _callable_alru_cache[(A, a: int, b: str), str] - correct
A.in_class(a, 1, '') # Argument of type "A" cannot be assigned to parameter of type "O@_wrapped_alru_cache" in function "__call__"
```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Pyright (pyright 1.1.187)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.188, which I just published. It will also be included in the next release of pylance.
